File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/billing/shared/bill-return-request.model.ts

import {
    NgForm,
    FormGroup,
    FormControl,
    Validators,
    FormBuilder,
    ReactiveFormsModule
} from '@angular/forms'

export class BillReturnRequest {
    public BillingTransactionItemId: number = 0;
    public BillingTransactionId: number = 0;

    public CounterId: number = 0;
    public PatientId: number = 0;
    public ItemName: string = null;
    public ItemId: number = 0;
    public ServiceDepartmentName: string = null;
    public ServiceDepartmentId: number = 0;
    public SubTotal: number = 0;
    public Price: number = 0;
    public DiscountAmount: number = 0;
    public Quantity: number = 0;
    public Tax: number = 0;
    public TotalAmount: number = 0;
    public ReturnDate: string = null;
    public PaidDate: string = null;
    public CreatedOn: string = null;
    public ReturnRemarks: string = null;
    public RequisitionId: number = 0;
    public CreatedBy: number = 0;
    public ProviderId: number = 0;

}

File: Code/Components/DanpheEMR.ServerModel/BillingModels/BillReturnRequestModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DanpheEMR.ServerModel
{
    public class BillReturnRequestModel
    {
        [Key]
        public int BillingReturnId { get; set; }
        public int BillingTransactionId { get; set; }
        public int BillingTransactionItemId { get; set; }
        public int CounterId { get; set; }
        public int PatientId { get; set; }
        public int ItemId { get; set; }
        public int ServiceDepartmentId { get; set; }
        public double? Price { get; set; }
        public double? SubTotal { get; set; }
        public double? DiscountAmount { get; set; }
        public double? Quantity { get; set; }
        public double? Tax { get; set; }
        public double? TotalAmount { get; set; }
        public DateTime ReturnDate { get; set; }
        public DateTime? CreatedOn { get; set; }
        public string ReturnRemarks { get; set; }
        //this will be employeeid of the current user..
        public int? CreatedBy { get; set; }
        public int? ProviderId { get; set; }
    }
}
